RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -1,32 +1,124 @@
 package com.rfs.common;
 
+import java.time.Duration;
+import java.util.Base64;
+import java.util.List;
+
+import io.netty.buffer.Unpooled;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.lucene.document.Document;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+import reactor.util.retry.Retry;
 
 
 public class DocumentReindexer {
     private static final Logger logger = LogManager.getLogger(DocumentReindexer.class);
+    private static final int MAX_BATCH_SIZE = 1000; // Arbitrarily chosen
+
+    public static void reindex(String indexName, Flux<Document> documentStream, ConnectionDetails targetConnection) throws Exception {
+        String targetUrl = "/" + indexName + "/_bulk";
+        HttpClient client = HttpClient.create()
```

Comment:
Can we split the http client into a separate class which may get reused by different operations

Reply:
I'd prefer not to at this point, actually.  While I would traditionally agree with you, I'm trying out a new approach on this project and been really happy with how it has worked out.  Specifically - avoiding being too speculative about abstractions and letting the needs of the project shape what gets created.  In this case, we only have one thing that needs this reactor-netty client, and I honestly don't know what interface I would provide if I were to carve it out because I don't know how another potential part of the code might use it.  Avoiding speculation on past abstractions in this project's history has been one of the key things that has enabled me to make so much progress so fast.

Reply:
I don't know if you need a separate HttpClient interface yet, but I do think that it might help & in general, you'll want to look to the future and not think about the past.
From my view, you've got some leaky abstractions with a couple other needless Flux contaminations within your codebase.
Once you do that, it will become harder to test your code too (less tests help us write application code faster too).  If you want to write test code fast too - keep it as generic as you can with the cleanest interfaces that you can strive for.  Simpler pieces -> smoother integrations -> faster delivery of quality solutions.